exercise-book/src/iterators.md

```diff
@@ -0,0 +1,324 @@
+# Iterators
+
+In this exercise, you will learn to manipulate and chain iterators. Iterators are a functional way to write loops and control flow logic.
+
+## After completing this exercise you are able to
+
+- write a Rust iterator
+- use closures in iterator chains
+- collect a result to different containers
+- turn off rust-analyzer inlay hints
+
+## Prerequisites
+
+For completing this exercise you need to have
+
+- knowledge of control flow
+- how to write basic functions
+- know basic Rust types
+
+## Task
+
+- Add the odd numbers in the following string using an iterator chain
+
+```text
+//ignore everything that is not a number
+1
+2
+3
+4
+five
+6
+7
+∞
+9
+X
+```
+
+- Take the template in [exercise-templates/iterators](../../exercise-templates/iterators/) as a starting point.
```

Comment:
I would prefer to have a snippet of code right here instead. This way the participants can work on the exercise with only a single page.

Reply:
Sure thing.